solo-io/gloo
# Remove hardcoded namespace in tracing test v1.17.x

# Description

The gateway namespace was hard-coded in the test. This worked fine in this repository, but if the namespace of the test that was executing changed, the gateway could not be found. As a result, running this test in solo-projects (which uses a different namespace for the test installation) failed, which broke our nightly builds.

We fixed this by moving the gateway to a separate file and applying it with the `"-n", s.testInstallation.Metadata.InstallNamespace` option.

## API changes

None

## Code changes

Tests only

## CI changes

None

## Docs changes

None

# Context

See the issue link.

## Interesting decisions

None

## Testing steps

`git checkout v1.17.16` - modify the [test namespace](https://github.com/solo-io/gloo/blob/2c3b8fb368828fd7411869fc0eae1f12f8dab5f9/test/kubernetes/e2e/tests/edge_gw_test.go#L20) here to something else. Run it like so:

```
TARGET_ARCH=amd64 GOLANG_PROTOBUF_REGISTRATION_CONFLICT=ignore go test -v -timeout 600s ./test/kubernetes/e2e/tests -run '^TestGlooGatewayEdgeGateway$/Tracing' --count 1
```

The above command should fail.

Then checkout this branch and re-execute the above command (keeping the changed value for the namespace in `edge_gw_test.go`). The test should succeed. This, in my opinion, is sufficient to demonstrate that the tests should succeed in solo-projects.

## Notes for reviewers

# Checklist:
